Lib/test/test_webbrowser.py

```diff
@@ -265,6 +267,23 @@ def test_register_default(self):
     def test_register_preferred(self):
         self._check_registration(preferred=True)
 
+    def test_environment_variable_BROWSER_is_respected(self):
+        script_helper.assert_python_ok("-c", textwrap.dedent("""
+            import shutil, unittest
+            from test import support, test_webbrowser
+
+            def mock_which(path):
+                return True
+
+            class Test(test_webbrowser.ImportTest):
```

Comment:
I almost missed that this re-ran the `ImportTest` suite with `BROWSER` set.

I think that's OK as a test case, but it would be clearer as part of a dedicated `BrowserOverrideTest` class that's placed after `ImportTest` (and mentions in a class level comment that it re-runs the `ImportTest` cases in a subprocess with `BROWSER` set.